index.js

```diff
@@ -45,63 +53,48 @@ function disconnect() {
  * Connect to a mongodb
  *
  * @param {string} mongoConnectionString complete mongo connection string. e.g. user:password@mongodb:27017/database
- * @param {object} [options] mongo client options see link below
- * https://mongodb.github.io/node-mongodb-native/1.4/driver-articles/mongoclient.html?highlight=server#db-a-hash-of-options-at-the-db-level-overriding-or-adjusting-functionality-not-supported-by-the-url
+ * @param {object} [options] mongo client options; see README
  *
- * @returns {Boolean} If a new connection has been established
+ * @returns {Promise<Mongoose>} Returns the connection
  */
 function connect(mongoConnectionString, options = {}) {
   // Generate options
   options = { ...defaultOptions, ...options }
-  logger.info('**** mongoose-harakiri: Connecting with these options;', options)
+  logger.info('**** mongoose-harakiri: Connecting with these options:')
+  logger.info('**** mongoose-harakiri: ', options)
+
+  // Apply behavior options
+  if (options.hasOwnProperty('killProcessOnDisconnect')) {
+    killProcessOnDisconnect = options.killProcessOnDisconnect
```

Comment:
I'm also thinking about keeping the whole options object instead of keeping a local variable. This is a singleton anyway, we could even modify the defaultOptions, I think. What do you think?

Reply:
As we need to remove 'killProcessOnDisconnect' from the options before we pass it into mongoose (it produces a warning), we would have to clone and remove the custom mongoose-harakiri options every time we connect. I would prefer seperation into its own variable (or own object if we have more options) here.

Reply:
well, we would only need to create another clone while passing it to mongoose. deletion from the actual object that we save is not necessary.

Reply:
Changing the defaultOptions would pass settings from the first connection into every following connect where the options used the first connect one are not passed (fist connect { ssl: false }, the second connect with {} would result in unexpected behavior of ssl not being enabled). I propose saving the generated options (default + user options) as a variable as you suggested and keeping everything in there (see new push).

Reply:
but every following connect doesn't result in a new connection, right? this is effectively a singleton.